test/spec/unit/core/targeting_spec.js

```diff
@@ -326,4 +326,46 @@ describe('targeting tests', function () {
       });
     });
   });
+
+  describe('setTargetingForAst', function () {
+    let sandbox;
+    beforeEach(function() {
+      sandbox = sinon.createSandbox();
+      sandbox.stub(targetingInstance, 'resetPresetTargetingAST').callsFake(function() {});
+      sandbox.stub(window.apntag, 'setKeywords').callsFake(function() {});
+    });
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    function mockGetAllTargeting(adUnitCodes) {
+      adUnitCodes = Array.isArray(adUnitCodes) ? adUnitCodes : Array(adUnitCodes);
+      sandbox.stub(targetingInstance, 'getAllTargeting').callsFake(function() {
+        let mockTargeting = {};
+        for (let i = 0; i < Object.keys(adUnitCodes).length; i++) {
+          let key = Object.keys(adUnitCodes)[i];
+          mockTargeting[key] = {hb_bidder: 'appnexus'};
+        }
+        return mockTargeting;
+      });
+    }
+
+    it('should set single addUnit code', function() {
+      let adUnitCode = 'testdiv-abc-ad-123456-0';
+      mockGetAllTargeting(adUnitCode);
+      targetingInstance.setTargetingForAst(adUnitCode);
+      expect(targetingInstance.getAllTargeting.called).to.equal(true);
+      expect(targetingInstance.resetPresetTargetingAST.called).to.equal(true);
+      expect(window.apntag.setKeywords.called).to.equal(true);
+    });
+
+    it('should set array of addUnit codes', function() {
+      let adUnitCodes = ['testdiv1-abc-ad-123456-0', 'testdiv2-abc-ad-123456-0']
+      mockGetAllTargeting(adUnitCodes);
+      targetingInstance.setTargetingForAst(adUnitCodes);
+      expect(targetingInstance.getAllTargeting.called).to.equal(true);
+      expect(targetingInstance.resetPresetTargetingAST.called).to.equal(true);
+      expect(window.apntag.setKeywords.called).to.equal(true);
```

Comment:
To differentiate between this test and the one with single adUnitCode, you should test here that `window.apntag.setKeywords` executed twice. Also another assertion can be called with args. Check this https://github.com/sinonjs/sinon/issues/953#issuecomment-265447178

Reply:
updated the tests.